ecosystem/typescript/sdk/src/providers/aptos_client.ts

```diff
@@ -308,6 +316,91 @@ export class AptosClient {
+    options?: Partial<Gen.SubmitTransactionRequest>,
+  ): Promise<TxnBuilderTypes.FeePayerRawTransaction> {
+    const rawTxn = await this.generateTransaction(sender, payload, options);
+
+    const signers: Array<AccountAddress> = secondary_signer_accounts.map((account) =>
+      AccountAddress.fromHex(account.address()),
+    );
+
+    const feePayerTxn = new TxnBuilderTypes.FeePayerRawTransaction(
+      rawTxn,
+      signers,
+      AccountAddress.fromHex(fee_payer.address()),
+    );
+    return feePayerTxn;
+  }
+
+  /**
+   * Submits fee payer transaction to chain
+   *
+   * @param feePayerTransaction the raw transaction to be submitted, of type FeePayerRawTransaction
+   * @param senderAuthenticator the sender account authenticator (can get from signMultiTransaction() method)
+   * @param feePayerAuthenticator the feepayer account authenticator (can get from signMultiTransaction() method)
+   * @param signersAuthenticators an optional array of the signer account authenticators
+   * @returns The pending transaction
+   */
+  async submitFeePayerTransaction(
+    feePayerTransaction: TxnBuilderTypes.FeePayerRawTransaction,
+    senderAuthenticator: TxnBuilderTypes.AccountAuthenticatorEd25519,
+    feePayerAuthenticator: TxnBuilderTypes.AccountAuthenticatorEd25519,
+    signersAuthenticators: Array<TxnBuilderTypes.AccountAuthenticatorEd25519> = [],
+  ): Promise<Gen.PendingTransaction> {
+    const txAuthenticatorFeePayer = new TxnBuilderTypes.TransactionAuthenticatorFeePayer(
+      senderAuthenticator,
+      feePayerTransaction.secondary_signer_addresses,
+      signersAuthenticators,
+      { address: feePayerTransaction.fee_payer_address, authenticator: feePayerAuthenticator },
+    );
+
+    const bcsTxn = bcsToBytes(
+      new TxnBuilderTypes.SignedTransaction(feePayerTransaction.raw_txn, txAuthenticatorFeePayer),
+    );
+    const transactionRes = await this.submitSignedBCSTransaction(bcsTxn);
+
+    return transactionRes;
+  }
+
+  /**
+   * Signs a multi transaction type (multi agent / fee payer) and returns the
+   * signer authenticator to be used to submit the transaction.
+   *
+   * @param signer the account to sign on the transaction
+   * @param rawTxn a MultiAgentRawTransaction or FeePayerRawTransaction
+   * @returns signer authenticator
+   */
+  // eslint-disable-next-line class-methods-use-this
+  async signMultiTransaction(
+    signer: AptosAccount,
+    rawTxn: MultiAgentRawTransaction | FeePayerRawTransaction,
+  ): Promise<TxnBuilderTypes.AccountAuthenticatorEd25519> {
+    const signerSignature = new TxnBuilderTypes.Ed25519Signature(
+      signer.signBuffer(TransactionBuilder.getSigningMessage(rawTxn)).toUint8Array(),
+    );
+
+    const signerAuthenticator = new TxnBuilderTypes.AccountAuthenticatorEd25519(
+      new TxnBuilderTypes.Ed25519PublicKey(signer.signingKey.publicKey),
+      signerSignature,
+    );
+
+    return Promise.resolve(signerAuthenticator);
+  }
```

Comment:
We could just make this `signTransaction` for any transaction right?  It should be the same either way?

Reply:
We already have a `signTransaction` that signs with `Ed25519Signature` and returns a`Uint8Array` type, for feepayer/multiagent transaction we need to return the signer authenticator so users would submit the transaction with it (unless we want to introduce another function in the flow that return the authenticator, which i dont think it is necessary) . because of different return type issues, we might be better with 2 separate functions

Reply:
I see... then let's keep this how it is.  We can do better when we rework the whole thing in V2.